src/frontend/sharedComponents/OfflineAreaCard.tsx

```diff
@@ -0,0 +1,161 @@
+    defaultMessage: "Global",
+    description: "Level of detail seen on map - farthest zoom level",
+  },
+  lvlDetailSubcontinent: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailSubcontinent",
+    defaultMessage: "Subcontinent",
+    description: "Level of detail seen on map - second farthest zoom level",
+  },
+  lvlDetailLargeCountry: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailLargeCountry",
+    defaultMessage: "Large Country",
+    description: "Level of detail seen on map - can see large countries",
+  },
+  lvlDetailSmallCountry: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailSmallCountry",
+    defaultMessage: "Small Country",
+    description: "Level of detail seen on map - can see smaller countries",
+  },
+  lvlDetailLargeMetro: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailLargeMetro",
+    defaultMessage: "Large Metropolitan Area",
+    description:
+      "Level of detail seen on map - Can see details of large metropolitan areas",
+  },
+  lvlDetailCity: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailCity",
+    defaultMessage: "City",
+    description: "Level of detail seen on map - Can see most Cities on Map",
+  },
+  lvlDetailTown: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailTown",
+    defaultMessage: "Town",
+    description: "Level of detail seen on map - can see most towns on map",
+  },
+  lvlDetailVillage: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailVillage",
+    defaultMessage: "Village",
+    description: "Level of detail seen on map - can see villages on map",
+  },
+  lvlDetailSmallRoad: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailSmallRoad",
+    defaultMessage: "Small Road",
+    description: "Level of detail seen on map - cab see small roads on map",
+  },
+  lvlDetailStreet: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailStreet",
+    defaultMessage: "Street",
+    description: "Level of detail seen on map - can see most streets on map",
+  },
+  lvlDetailStreetBlock: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailStreetBlock",
+    defaultMessage: "Street Block",
+    description:
+      "Level of detail seen on map - can see details of street blocks on map",
+  },
+  lvlDetailAddress: {
+    id: "sharedComponent.OfflineAreaCard.lvlDetailAddress",
+    defaultMessage: "Address",
+    description: "Level of detail seen on map - Can see addresses on map",
+  },
+  lvlDetailStreetIntersection: {
+    id: "sharedComponent.OfflineAreaCard.StreetIntersection",
+    defaultMessage: "Street Intersection",
+    description:
+      "Level of detail seen on map - can see street intersections on map",
+  },
+});
+
+interface OfflineAreaCardProps {
+  title: string;
+  zoomLevel: number;
+  position: number;
+}
+
+export const OfflineAreaCard = ({
+  title,
+  zoomLevel,
+  position,
+}: OfflineAreaCardProps) => {
+  const { formatMessage: t } = useIntl();
+
+  const lvlOfDetail = React.useMemo(() => {
```

Comment:
the memoization is probably not necessary here. would also extract this into a function outside the component

Reply:
hm i guess a little bit more complicated to extract since it relies on the format message function. maybe extract the logic so that it returns the appropriate message key name e.g.

```jsx
// Not the best name but you get the idea
function getLevelOfDetailKey(zoomLevel: number) {
  switch (true) {
    case zoomLevel <= 2:
      return 'lvlDetailGlobal';
    // ...
  }

const Component = () => {
  // ...
  const lvlOfDetail = t(m[getLevelOfDetailKey(zoomLevel)])
  // ... 
}